cpp/src/gandiva/precompiled/extended_math_ops.cc

```diff
@@ -367,4 +367,33 @@ gdv_float64 get_scale_multiplier(gdv_int32 scale) {
   return power_float64_float64(10.0, scale);
 }
 
+// convert input unsigned long to its binary representation
+void bin(uint64_t n, char* ret, int32_t* position) {
+  if (n > 1) {
+    bin(n / 2, ret, position);
+  }
+  ret[*position] = (n % 2) == 0 ? '0' : '1';
+  *position += 1;
+}
+
+// returns the binary representation of a given integer (e.g. 928 -> 1110100000)
+#define BIN_INTEGER(IN_TYPE)                                                          \
+  FORCE_INLINE                                                                        \
+  const char* bin_##IN_TYPE(int64_t context, gdv_##IN_TYPE value, int32_t* out_len) { \
```

Comment:
It doesn't work with negative integers

Reply:
This function was built to have the same behavior as Hive BIN function. 
Can you be more clear in what do you consider that won't work?

If you check the tests, there are cases with negative int values working and behaving correctly as Hive.

Reply:
the negative ints output as 64 chars instead of 32 chars. I checked now with hive and it seems they are also doing like this. I think its better to keep it as 32 sized.